history/history-network.md

```diff
@@ -25,549 +22,157 @@ In addition, the chain history network provides block number to historical block
-* The client accepting an Offer **MUST** random gossip the headers from the accepted anchored header to the end of the content keys list, even if the client only accepts a subset of the respective range.
-
-Validation Logic
-* When processing accepted headers, if the requested range cannot be validated by walking backward through the header's `parent_hash`'s, clients MAY penalize or descore the peer.
-* The anchored `block_hash`'s corresponding header can be validated via `keccak256(rlp.encode(header)) == block_hash`
-* The ancestors can be validated by recursively walking backwards and verifying `keccak256(rlp.encode(parent_header)) == current_header.parent_hash`
+The History Network contains block body and receipts content types. These types use `rlp` encoding of the underlying objects.
 
 #### Block Body
 
-After the addition of `withdrawals` to the block body in the [EIP-4895](https://eips.ethereum.org/EIPS/eip-4895),
-clients need to support multiple encodings for the block body content type. For the time being,
-since a client is required for block body validation it is recommended that clients implement
-the following sequence to decode & validate block bodies.
-
-- Receive raw block body content value.
-- Fetch respective header from the network.
-- Compare header timestamp against `SHANGHAI_TIMESTAMP` to determine what encoding scheme the block body uses.
-- Decode the block body using either pre-shanghai or post-shanghai encoding.
-- Validate the decoded block body against the roots in the header.
+The Block body content type contains block's transactions, ommers and withdrawals.
 
 ```python
-block_body_key          = Container(block_hash: Bytes32)
-selector                = 0x01
-
-# Transactions
-transactions            = List(ssz_transaction, limit=MAX_TRANSACTION_COUNT)
-ssz_transaction         = ByteList[MAX_TRANSACTION_LENGTH](encoded_transaction)
-encoded_transaction     =
-  if transaction.is_typed:
-    return transaction.type_byte + rlp.encode(transaction)
-  else:
-    return rlp.encode(transaction)
-
-# Uncles
-uncles                  = ByteList[MAX_ENCODED_UNCLES_LENGTH](encoded_uncles)
-encoded_uncles          = rlp.encode(list_of_uncle_headers)
-
-# Withdrawals
-withdrawals             = List(ssz_withdrawal, limit=MAX_WITHDRAWAL_COUNT)
-ssz_withdrawal          = ByteList[MAX_WITHDRAWAL_LENGTH](encoded_withdrawal)
-encoded_withdrawal      = rlp.encode(withdrawal)
-
-# Block body
-pre_shanghai_body       = Container(
-  transactions: transactions,
-  uncles: uncles
-)
-post_shanghai_body      = Container(
-  transactions: transactions,
-  uncles: uncles,
-  withdrawals: withdrawals
-)
-
-# Encoded content
-content                 = SSZ.serialize(pre_shanghai_body | post_shanghai_body)
-content_key             = selector + SSZ.serialize(block_body_key)
-```
+selector        = 0x09
```

Comment:
We could also decide to reset these to 0x00 and 0x01 for bodies & receipts.

Reply:
Note that this would not even conflict with existing legacy networks as the Protocol ID is still different. So we can definitely start from 0x00 and 0x01